connectors/salesforce/src/main/java/com/vmware/connectors/salesforce/SalesforceController.java

```diff
@@ -737,4 +741,48 @@ private String formatRecipient(UserRecord userRecord) {
                 .bodyToMono(String.class);
     }
 
+    @PostMapping(
+            path = UPDATE_CLOSE_DATE,
+            consumes = APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Mono<Void> updateCloseDate(
+            @RequestHeader(SALESFORCE_AUTH_HEADER) final String auth,
+            @RequestHeader(SALESFORCE_BASE_URL_HEADER) final String baseUrl,
+            @PathVariable(OPPORTUNITY_ID) final String opportunityId,
+            @RequestParam("closedate") final String closeDate) {
+
+        // CloseDate should in the format "YYYY-MM-DD".
+        final Map<String, String> body = ImmutableMap.of("CloseDate", closeDate);
+
+        return updateOpportunityField(baseUrl, auth, opportunityId, body);
+    }
+
+    @PostMapping(
+            path = UPDATE_NEXT_STEP,
+            consumes = APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Mono<Void> updateNextStep(
+            @RequestHeader(SALESFORCE_AUTH_HEADER) final String auth,
+            @RequestHeader(SALESFORCE_BASE_URL_HEADER) final String baseUrl,
+            @PathVariable(OPPORTUNITY_ID) final String opportunityId,
+            @RequestParam("nextstep") final String nextStep
+    ) {
+
+        final Map<String, String> body = ImmutableMap.of("NextStep", nextStep);
+
+        return updateOpportunityField(baseUrl, auth, opportunityId, body);
+    }
+
+    private Mono<Void> updateOpportunityField(final String baseUrl,
+                                              final String auth,
+                                              final String opportunityId,
+                                              final Object body) {
+        return rest.patch()
+                .uri(baseUrl + "/services/data/v39.0/sobjects/Opportunity/{opportunityId}", opportunityId)
```

Comment:
We are using v20.0 for all the existing functions.
Does any of these actions needs to bump the version ?

https://github.com/vmware/connectors-workspace-one/blob/master/connectors/salesforce/src/main/resources/application.yml

Reply:
When the existing functionalities were built,```v20.0``` would have been the latest. Currently, ```v39.0``` is the API latest version. It is always better to go for latest API version.

Reply:
+1 for updating all of the versions to the latest (be sure to manually test that it all still works)

Also, to keep everything in the same place and able to be overridden (either everything should be overridable or none of it should be imo), move the url into the yml like the others.

Reply:
I have moved the API to ```application.yml``` file. I have to go through the salesforce API document to see if there are any side-effects of upgrading the existing APIs to the latest version.

I have created a follow-up ticket APF-1159 to update the existing APIs to ```v39.0```.